tools/sdk-testgen/packages/autorest.gotest/doc/gotest_howto.md

```diff
@@ -0,0 +1,156 @@
+
+1. Use `@autorest/gotest` to generate the example code. If you want to generate the SDK code as well, add `--generate-sdk` tag.
+```sh
+autorest --version=3.8.2 --use=@autorest/go@latest --use=@autorest/gotest@latest --go --track2 --output-folder=<sdk-repo-workspace>/sdk/resourcemanager/signalr/armsignalr --clear-output-folder=false --go.clear-output-folder=false --testmodeler.generate-sdk-example <sdk-repo-workspace>/sdk/resourcemanager/signalr/armsignalr/autorest.md
+```
+2. Resolve dependency.
+```sh
+cd <sdk-repo-workspace>/sdk/resourcemanager/signalr/armsignalr
+go mod tidy
+```
+
+## Generate mock test and execute
+
+We use [swagger defined examples](https://github.com/Azure/azure-rest-api-specs/blob/main/document/x-ms-examples.md) to generate mock tests. These tests can be executed with [mock service host](https://github.com/Azure/azure-sdk-tools/tree/main/tools/mock-service-host) to ensure the correctness of SDK.
+
+1.  Use `@autorest/gotest` to generate the mock tests. If you want to generate the SDK code as well, add `--generate-sdk` tag.
+```sh
+autorest --version=3.8.2 --use=@autorest/go@latest --use=@autorest/gotest@latest --go --track2 --output-folder=<sdk-repo-workspace>/sdk/resourcemanager/signalr/armsignalr --clear-output-folder=false --go.clear-output-folder=false --testmodeler.generate-mock-test <sdk-repo-workspace>/sdk/resourcemanager/signalr/armsignalr/autorest.md
+```
+2. Resolve dependency.
+```sh
+cd <sdk-repo-workspace>/sdk/resourcemanager/signalr/armsignalr
+go mod tidy
+```
+3. Create workspace for mock service host and install pacakge. You could reference the [readme](https://github.com/Azure/azure-sdk-tools/tree/main/tools/mock-service-host#readme) of mock service host for further usage.
+```sh
+mkdir <mock-service-host-workspace> && cd <mock-service-host-workspace>
+npm install @azure-tools/mock-service-host
+```
+4. Set mock service host config to use the same swagger with test generation.
+```sh
+cd <mock-service-host-workspace>
+echo "specRetrievalGitUrl=https://github.com/Azure/azure-sdk-tools
+validationPathsPattern=tools/sdk-testgen/swagger/specification/signalr/resource-manager/*/**/*.json" > .env
+```
+5. Start mock service host.
+```sh
+cd <mock-service-host-workspace>
+node node_modules/@azure-tools/mock-service-host/dist/src/main.js
+```
+6. Execute mock test and gather coverage result.
+```sh
+cd <sdk-repo-workspace>/sdk/resourcemanager/signalr/armsignalr
+go test -v -coverprofile coverage.txt
+```
+
+## Generate scenario test and execute
+
+We use [API scenarios](https://github.com/Azure/azure-rest-api-specs/tree/main/document/api-scenario) to generate scenario tests. These tests will use live traffic to ensure the consistency between SDK and service. We could use [test proxy](https://github.com/Azure/azure-sdk-tools/tree/main/tools/test-proxy) to record the live traffic and playback later without charging.

```

Comment:
Same above, can we simplified the step to do it within one command?

Reply:
Unfortunately, we do not have an all-in-one script for live test. All live test code will be committed to the repo. So, the pipeline will only start test-proxy and do the test. Also, this process is handled by the pipeline, not all in script.